static/js/legal/LegalSearch.js

```diff
@@ -9,24 +9,36 @@ const Pagination = require('./Pagination');
 class LegalSearch extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { q: $('#query').val(), from_hit: 0, advisory_opinions: []}
+    const initState = URI.parseQuery(window.location.search);
+    initState.q = initState.search;
+    initState.type = initState.search_type;
+    initState.advisory_opinions = [];
+    initState.from_hit = initState.from_hit ? parseInt(initState.from_hit, 10) : 0;
+    this.state = initState;
 
     this.getResults = this.getResults.bind(this);
     this.setQuery = this.setQuery.bind(this);
     this.instantQuery = this.instantQuery.bind(this);
     this.getResults();
   }
 
-  setQuery(e) {
+  setQuery(e, callback) {
     const value = e.target.type === 'checkbox' ? e.target.checked : e.target.value;
-    this.setState({ [e.target.name]: value, lastFilter: e.target.name});
+    const newState = {[e.target.name]: value, lastFilter: e.target.name};
+
+    if(e.target.name !== 'from_hit') {
```

Comment:
if any of the substantive searches change, I want to return results starting from page 1 (offset 0). But if `from_hit` is triggered, I want to increment / decrement as intended.

This came out of some testing I did where I went to the second page, then changed the filter, than was still at page 2 of the _new_ search, which seemed really strange to me.